swarm_dial.go

```diff
@@ -16,16 +16,39 @@ import (
-//  ------------------------>------------<-------     >---------<-----------------
-//  -----------------------|              \----x                 \----------------
-//  ----------------------|                \-----x                \---------------
-//                                         any may fail          if no addr at end
-//                                                             retry dialAttempt x
+// Concurrent Calls
+//  to Dial(peerId)                             dial finds dialable
+//       /                                      addresses and passes
+//       v                                        them to dialAddrs
+//  Dial(peerId) --------------------\                     |
+//        Dial(peerId)  -------------|                     v
+//          Dial(peerId) ------------>----> doDial() --> dial() --> dialAddrs() <- dialAddrs concurrently dials
+//          Dial(peerId) ------------| ^                              __/\__       all known peer addresses
+//                Dial(peerId) ------/ |                             /||||||\
+//                                    /                              ||||||||
+//                            Synced with dialsync                   LLLLL--- <- Costly paths are delayed
+//                              calls doDial                         DDDDD
+//                                                                   |||||
+//                                /--------------------------------> ||---  <- limitedDial (LD) limits
+//     Each connection is established                                ||        concurrency for transports
+//     by dialAddr, which starts a       Some connection attempts -> x|-       which require file descriptors
+//     network connection and sets               may fail         |   ||
+//     up the encryption layers                                   |   ||  --- <- if needed, dials through costly
+//                                                                |   ||  |||    paths (such as relay) will start
+//                                                                |   ||  LLL
+//                                                                \-> xx--DDD
+//                                                                      |||||
+//                                       eventually a connection will   |||||
+//                                        get established (C), other -> |C|||
+//       Connection setup will be        attempts will get cancelled    x|xxx
+//     performed with dialConnSetup.                                     |
+//     It adds the connection to the ----------------------------------> |
+//        swarm and agrees on the                                        |
+//            muxer to use                                             /---\
+//                                        The result is distributed -> |||||
+//                                            to callers of Dial       |||||
```

Comment:
@Stebalien after the transport refactor only the part after 'dialConnSetup' changed, right?

Reply:
Yes.